diff --git a/taskcluster/docker/snap-coreXX-build/run.sh b/taskcluster/docker/snap-coreXX-build/run.sh
--- a/taskcluster/docker/snap-coreXX-build/run.sh
+++ b/taskcluster/docker/snap-coreXX-build/run.sh
@@ -17,10 +17,11 @@
 
 export LC_ALL=C.UTF-8
 export LANG=C.UTF-8
 export SNAP_ARCH=amd64
 export SNAPCRAFT_BUILD_INFO=1
+export MOZILLA_SNAP_TEST=1
 
 export PATH=$PATH:$HOME/.local/bin/
 unset MOZ_AUTOMATION
 
 MOZCONFIG=mozconfig.in
diff --git a/toolkit/crashreporter/tools/symbolstore.py b/toolkit/crashreporter/tools/symbolstore.py
--- a/toolkit/crashreporter/tools/symbolstore.py
+++ b/toolkit/crashreporter/tools/symbolstore.py
@@ -802,14 +802,11 @@
 
             if "asan" not in perfherder_extra_options.lower():
                 print(
                     "PERFHERDER_DATA: %s" % json.dumps(perfherder_data), file=sys.stderr
                 )
-                if (
-                    "MOZ_AUTOMATION" in os.environ
-                    or "SNAPCRAFT_BUILD_INFO" in os.environ
-                ):
+                if "MOZ_AUTOMATION" in os.environ or "MOZILLA_SNAP_TEST" in os.environ:
                     upload_dir = Path(os.environ.get("UPLOAD_DIR"))
                     upload_dir.mkdir(parents=True, exist_ok=True)
                     upload_path = upload_dir / "perfherder-data-compiler-metrics.json"
                     with upload_path.open("w", encoding="utf-8") as f:
                         json.dump(perfherder_data, f)

