diff --git a/testing/profiles/profileserver/user.js b/testing/profiles/profileserver/user.js
index 19ff7d474f6d2..dbdec03fef03c 100644
--- a/testing/profiles/profileserver/user.js
+++ b/testing/profiles/profileserver/user.js
@@ -8,3 +8,5 @@
 user_pref("dom.timeout.enable_budget_timer_throttling", false);
 // Turn off update
 user_pref("app.update.disabledForTesting", true);
+// Bug 1925822: Snap build crashes during PGO step with GPU process
+user_pref("gfx.webrender.software", true);
