diff --git a/python/mozbuild/mozbuild/gn_processor.py b/python/mozbuild/mozbuild/gn_processor.py
--- a/python/mozbuild/mozbuild/gn_processor.py
+++ b/python/mozbuild/mozbuild/gn_processor.py
@@ -15,11 +15,11 @@
 
 from mozbuild.backend.base import BuildBackend
 import mozpack.path as mozpath
 from mozbuild.frontend.sandbox import alphabetical_sorted
 from mozbuild.frontend.data import GnProjectData
-from mozbuild.util import expand_variables, mkdir
+from mozbuild.util import mkdir
 
 
 license_header = """# This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
@@ -323,16 +323,10 @@
                 ".mm": "CMMFLAGS",
             }
             variables = (suffix_map[e] for e in extensions if e in suffix_map)
             for var in variables:
                 for f in flags:
-                    # We may be getting make variable references out of the
-                    # gn data, and we don't want those in emitted data, so
-                    # substitute them with their actual value.
-                    f = expand_variables(f, config.substs).split()
-                    if not f:
-                        continue
                     # the result may be a string or a list.
                     if isinstance(f, six.string_types):
                         context_attrs.setdefault(var, []).append(f)
                     else:
                         context_attrs.setdefault(var, []).extend(f)

