diff --git a/python/mozbuild/mozbuild/gn_processor.py b/python/mozbuild/mozbuild/gn_processor.py
--- a/python/mozbuild/mozbuild/gn_processor.py
+++ b/python/mozbuild/mozbuild/gn_processor.py
@@ -176,12 +176,12 @@
         "MOZ_DEBUG": "1" if input_vars.get("is_debug") else None,
         "OS_TARGET": oses[input_vars["target_os"]],
         "HOST_CPU_ARCH": cpus.get(input_vars["host_cpu"], input_vars["host_cpu"]),
         "CPU_ARCH": cpus.get(input_vars["target_cpu"], input_vars["target_cpu"]),
     }
-    if input_vars["target_os"] in ("linux", "android", "openbsd"):
-        mozbuild_args["MOZ_X11"] = "1" if input_vars.get("use_x11") else None
+    if "use_x11" in input_vars:
+        mozbuild_args["MOZ_X11"] = "1" if input_vars["use_x11"] else None
 
     gn_out["mozbuild_args"] = mozbuild_args
     all_deps = find_deps(gn_result["targets"], gn_target)
 
     for target_fullname in all_deps:

