diff --git a/dom/media/webrtc/third_party_build/gn-configs/README-openbsd.md b/dom/media/webrtc/third_party_build/gn-configs/README-openbsd.md
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/README-openbsd.md
+++ /dev/null
@@ -1,73 +0,0 @@
-# Setup a Virtual Machine running OpenBSD to generate OpenBSD build files for libwebrtc.
-
-1. Download the OpenBSD 7.0 ISO [here](https://cdn.openbsd.org/pub/OpenBSD/7.0/amd64/install70.iso)
-
-2. Start by setting up a new VM. These steps reference VMWare fusion options,
-   but they should translate to other VM hosts.  During the VM creation steps,
-   using the following settings:
-   - VMWare - Other Linux 5.x kernel 64-bit
-   - VMWare - Legacy BIOS
-   - VMWare - Customize settings
-   - VMWare - Processors & Memory - 2 cores, 2048MB Memory
-   - VMWare - Hard Disk - 50.00 GB
-
-3. Start VM to begin the OpenBSD install process.
-
-   In general, choose the default options for everything unless listed below:
-   - select ```(I)nstall```
-   - allow root ssh - ```yes```
-   - disk - for more info see [here](https://azcrumpty.weebly.com/journal/easy-openbsd-partition-scheme)
-       - ```c``` # for custom
-       - ```z``` # for delete all
-       - ```a a``` # for add 'a' partition
-       - enter forward slash ('/') for the mount point
-       - ```q``` # to quit editor
-   - directory does not contain SHA256.sig. Continue without verification? ```yes```
-
-4. Upgrade to OpenBSD -current
-
-   Upgrading to -current is required for the minimum rust version.  For more
-   info, see [this](https://unix.stackexchange.com/questions/406870/how-to-follow-openbsd-current).
-
-   - login as root
-
-           ftp -o /bsd.rd https://cloudflare.cdn.openbsd.org/pub/OpenBSD/snapshots/amd64/bsd.rd
-           reboot
-
-   - at the bootloader prompt _quickly_ type
-
-           boot bsd.rd
-
-   - select ```U(Upgrade)``` at first prompt question, default to all other questions
-   - after upgrade completes allow reboot
-
-5. Upgrade and install necessary packages, add non-root user
-
-   - login as root
-
-         pkg_add -u
-         pkg_add sudo-- git-- bash-- mercurial-- m4-- py3-pip--
-         ln -sf /usr/local/bin/pip3.9 /usr/local/bin/pip3
-         ln -sf /usr/local/bin/python3 /usr/local/bin/python
-         visudo # uncomment line with '%wheel        ALL=(ALL) NOPASSWD: SETENV: ALL'
-
-         export NEW_USER={pick-a-user-name}
-         useradd -b /home -m -G wheel $NEW_USER
-         usermod -s /usr/local/bin/bash $NEW_USER
-         passwd $NEW_USER
-
-   - logout
-
-6. Clone mozilla-central repository and bootstrap
-
-   - login as {NEW_USER}
-
-           hg clone https://hg.mozilla.org/mozilla-central
-           (cd mozilla-central && ./mach bootstrap)
-
-   - for bootstrap:
-       - ignoring errors around glean-sdk
-       - select option ```1``` for all pkg_add prompts
-
-
-7. Continue with step 4 in README.md
diff --git a/dom/media/webrtc/third_party_build/gn-configs/README.md b/dom/media/webrtc/third_party_build/gn-configs/README.md
--- a/dom/media/webrtc/third_party_build/gn-configs/README.md
+++ b/dom/media/webrtc/third_party_build/gn-configs/README.md
@@ -1,85 +1,31 @@
 # Generate new gn json files and moz.build files for building libwebrtc in our tree
 
-1. If generating on macOS for Apple Silicon (cross-compiling), make sure to have
-   at least [Xcode 12.2](https://download.developer.apple.com/Developer_Tools/Xcode_12.2/Xcode_12.2.xip).
-
-   The aarch64 Rust target will need to be installed via:
-   `rustup target add aarch64-apple-darwin`
-
-2. If generating on Windows 10, Visual Studio 2019 is required.  Please follow
-   the install instructions from [here](https://firefox-source-docs.mozilla.org/setup/windows_build.html)
+/!\ This is only supported on Linux and macOS. If you are on Windows, you can run
+the script under [WSL](https://docs.microsoft.com/en-us/windows/wsl/install).
 
-   In addition, the following options must be selected in the VS2019 installer:
-   - C++ ATL for latest v142 build tools (ARM64)
-   - Windows 10 SDK (10.0.19041.0)
-   - MSVC v142 - VS2019 C++ ARM64 build tools
-
-   "Debugging Tools for Windows" is also required.
-   - Under `Settings -> Apps` search for "Windows Software Development Kit" with
-     the version number 10.0.19041.685.
-   - select Modify (and allow the installer to modify)
-   - select Change and then click Next
-   - select "Debugging Tools for Windows" and then click Change.
-
-   The aarch64 Rust target will need to be installed via:
-   `rustup target add aarch64-pc-windows-msvc`
-
-3. Download a version of the `gn` executable that corresponds to
+1. Download a version of the `gn` executable that corresponds to
     `Thu Nov 19 14:14:00 2020`.  In our case, that is version `1889 (8fe02009)`.
    
-   - [Win](https://chrome-infra-packages.appspot.com/p/gn/gn/windows-amd64/+/e_UmTHedzuu4zJ2gdpW8jrFFNnzIhThljx3jn3RMlVsC)
    - [Linux](https://chrome-infra-packages.appspot.com/p/gn/gn/linux-amd64/+/bvBFKgehaepiKy_YhFnbiOpF38CK26N2OyE1R1jXof0C)
    - [macOS](https://chrome-infra-packages.appspot.com/p/gn/gn/mac-amd64/+/nXvMRpyJhLhisAcnRmU5s9UZqovzMAhKAvWjax-swioC)
 
    Find the downloaded `.zip` file, unzip and export the location of the
    executable:
 
         unzip gn-mac-amd64.zip && export GN=`pwd`/gn
-        unzip gn-windows-amd64.zip && export GN=`pwd`/gn.exe
         unzip gn-linux-amd64.zip && export GN=`pwd`/gn
 
-   On platforms that don't have pre-built `gn` executables, `ninja` and `gn` can
-   be easily built:
-
-        git clone https://github.com/ninja-build/ninja.git
-        git clone https://gn.googlesource.com/gn
-        (cd gn && git checkout 8fe02009)
-        (cd ninja && ./configure.py --bootstrap)
-        (export NINJA=`pwd`/ninja/ninja ; cd gn && python build/gen.py && $NINJA -C out)
-        export GN=`pwd`/gn/out/gn
-
-   On OpenBSD, a slightly newer version of `gn` is needed in order to build:
-
-        (cd gn && git checkout 31f2bba8)
-
-4. It is time to generate the build files.  The script should be run from the
+2. It is time to generate the build files.  The script should be run from the
    top directory of our firefox tree.
 
         bash ./dom/media/webrtc/third_party_build/gn-configs/generate-gn-build-files.sh
 
    Debugging the generate script itself may prove useful, and one can do this by
    setting the DEBUG_GEN environment variable to a non-empty value. This will 
    print everything that the script executes.
 
-5. Checkin all the generated/modified files and try your build!
+3. Checkin all the generated/modified files and try your build!
 
 # Adding new configurations to the build
 
-- Each new platform/architecture will require 2 new mozconfig files, one for the
-  debug build and one for the non-debug build.  The filenames follow the same
-  pattern as the generated json files, `a-b-c-d.mozconfig` where:
-  - a = generating cpu (example: x64)
-  - b = debug (True / False)
-  - c = target cpu (example: x64 / arm64)
-  - d = target platform (mac/linux)
-- Each mozconfig file defines, in addition to debug/non-debug, the output
-  directory (MOZ_OBJDIR) and any architecture definition.
-- The new configs must be added to the appropriate platform section in
-  `generate-gn-build-files.sh`.
-
-**Note:** when adding new mozconfig files, especially for linux/android configs,
-it is important to include the `ac_add_options --enable-bootstrap`.  This
-ensures switching archtectures for "cross-compiled" generation works properly.
-For example, when generating `x86` or `arm64` linux json files, it would be
-necessary to install additional libraries in order for the configure step to
-complete.
+Edit the `GnConfigGenBackend` class in `python/mozbuild/mozbuild/gn_processor.py` file.
diff --git a/dom/media/webrtc/third_party_build/gn-configs/fixup_json.py b/dom/media/webrtc/third_party_build/gn-configs/fixup_json.py
--- a/dom/media/webrtc/third_party_build/gn-configs/fixup_json.py
+++ b/dom/media/webrtc/third_party_build/gn-configs/fixup_json.py
@@ -21,24 +21,23 @@
     f = open(filein, "r")
     file = f.read()
 
     # Remove references to CR_XCODE_VERSION and the output directory of the gn process.
     # This change does not cause a change in the generated moz.build files.
-    file = re.sub(r' *"CR_XCODE_VERSION=[0-9]+",\n', r"", file)
+    file = re.sub(r' *"CR_XCODE_VERSION=([0-9.]+)",\n', r"", file)
     file = re.sub(r' *"CR_SYSROOT_HASH=[0-9a-f]+",\n', r"", file)
     file = re.sub(r',\n *"(.\:)?/.*/third_party/libwebrtc/gn-output/gen/"', r"", file)
 
     # In practice, almost all of the entries in the cflags section have no affect
     # on the moz.build output files when running ./mach build-backend -b GnMozbuildWriter
     # There are few exceptions which do: -msse2, -mavx2, -mfma, -fobjc-arc
     # However, since we're really concerned about removing differences between development
     # machines, we only need remove the reference to osx sdk.  Removing it doesn't change
     # the generated moz.build files and makes diffs much easier to see.
     file = re.sub(
-        r' *"-isysroot",\n *"[\./]*/Applications/Xcode\.app/Contents'
-        "/Developer/Platforms/MacOSX\.platform/Developer/SDKs/"
-        'MacOSX([0-9][0-9]\.[0-9])?\.sdk",\n',
+        r' *"-isysroot",\n *".*/Contents/Developer/Platforms/MacOSX\.platform/Developer/SDKs/'
+        'MacOSX([0-9]+\.[0-9]+)?\.sdk",\n',
         r"",
         file,
     )
     f.close()
 
diff --git a/dom/media/webrtc/third_party_build/gn-configs/generate-gn-build-files.sh b/dom/media/webrtc/third_party_build/gn-configs/generate-gn-build-files.sh
--- a/dom/media/webrtc/third_party_build/gn-configs/generate-gn-build-files.sh
+++ b/dom/media/webrtc/third_party_build/gn-configs/generate-gn-build-files.sh
@@ -38,90 +38,27 @@
   echo "There are modified files in the checkout. Cowardly aborting!"
   echo "$MODIFIED_FILES"
   exit 1
 fi
 
-IS_WIN=0
-IS_DARWIN=0
-IS_LINUX=0
-
-# Erase content of third_party/libwebrtc/moz.build to help with generating
-# files that may conflict with current moz.build files.  Each config calls
-# ./mach configure.  If there are conflicts in any moz.build files under
-# third_party/libwebrtc, it would keep our configure step from completing
-# successfully.  Since this file will be regenerated at the end of this
-# process, we can make it an empty file to avoid conflicts during the
-# configure step.
-echo "" > third_party/libwebrtc/moz.build
-
-# For now, only macOS, Windows, and Linux (including Android builds) are supported here.
-if [ "x$SYS_NAME" = "xDarwin" ]; then
-  CONFIGS="x64_False_arm64_mac x64_True_arm64_mac x64_False_x64_mac x64_True_x64_mac"
-  IS_DARWIN=1
-elif [ "x$SYS_NAME" = "xMINGW32_NT-6.2" ]; then
-  unset ANSICON
-  CONFIGS="x64_True_arm64_win x64_False_arm64_win"
-  CONFIGS="$CONFIGS x64_True_x64_win x64_False_x64_win"
-  CONFIGS="$CONFIGS x64_True_x86_win x64_False_x86_win"
-  IS_WIN=1
-elif [ "x$SYS_NAME" = "xOpenBSD" ]; then
-  CONFIGS="x64_False_x64_openbsd x64_True_x64_openbsd"
-else
-  # Ensure rust has the correct targets for building x86 and arm64.  These
-  # operations succeed quickly if previously completed.
-  rustup target add aarch64-unknown-linux-gnu
-  rustup target add i686-unknown-linux-gnu
-
-  CONFIGS="x64_False_x64_linux_False x64_True_x64_linux_False"
-  CONFIGS="$CONFIGS x64_False_x64_linux_True x64_True_x64_linux_True"
-  CONFIGS="$CONFIGS x64_False_x86_linux_False x64_True_x86_linux_False"
-  CONFIGS="$CONFIGS x64_False_x86_linux_True x64_True_x86_linux_True"
-  CONFIGS="$CONFIGS x64_False_arm64_linux_False x64_True_arm64_linux_False"
-  CONFIGS="$CONFIGS x64_False_arm64_linux_True x64_True_arm64_linux_True"
-  CONFIGS="$CONFIGS x64_False_arm_android x64_True_arm_android"
-  CONFIGS="$CONFIGS x64_False_x64_android x64_True_x64_android"
-  CONFIGS="$CONFIGS x64_False_x86_android x64_True_x86_android"
-  CONFIGS="$CONFIGS x64_False_arm64_android x64_True_arm64_android"
-  IS_LINUX=1
-fi
-
 CONFIG_DIR=dom/media/webrtc/third_party_build/gn-configs
 echo "CONFIG_DIR is $CONFIG_DIR"
 
-# Each mozconfig is for a particular "platform" and defines debug/non-debug and a
-# MOZ_OBJDIR based on the name of the mozconfig to make it easier to know where to
-# find the generated gn json file.
-# The output of this step is a series of gn produced json files with a name format
-# that follows the form a-b-c-d.json where:
-#   a = generating cpu (example: x64)
-#   b = debug (True / False)
-#   c = target cpu (example: x64 / arm64)
-#   d = target platform (mac/linux)
-for THIS_BUILD in $CONFIGS
-do
-  echo "Building gn json file for $THIS_BUILD"
-  export MOZCONFIG=$CONFIG_DIR/$THIS_BUILD.mozconfig
-  echo "Using MOZCONFIG=$MOZCONFIG"
-
-  ./mach configure | tee $THIS_BUILD.configure.log
-  if [ ! -d obj-$THIS_BUILD ]; then
-    echo "Expected build output directory obj-$THIS_BUILD is missing, ensure this is set in $MOZCONFIG"
-    exit 1
-  fi
-  ./mach build-backend -b GnConfigGen --verbose | tee $THIS_BUILD.build-backend.log
-  cp obj-$THIS_BUILD/third_party/libwebrtc/gn-output/$THIS_BUILD.json $CONFIG_DIR
-done
+export MOZ_OBJDIR=$(mktemp -d -p . obj-XXXXXXXXXX)
+./mach configure
+if [ ! -d $MOZ_OBJDIR ]; then
+  echo "Expected build output directory $MOZ_OBJDIR is missing"
+  exit 1
+fi
+./mach build-backend -b GnConfigGen --verbose | tee $MOZ_OBJDIR/build-backend.log
+cp $MOZ_OBJDIR/third_party/libwebrtc/gn-output/*.json $CONFIG_DIR
 
 # run some fixup (mostly removing dev-machine dependent info) on json files
-for THIS_CONFIG in $CONFIGS
+for THIS_CONFIG in $CONFIG_DIR/*.json
 do
-  echo "fixup file: $CONFIG_DIR/$THIS_CONFIG.json"
-  ./$CONFIG_DIR/fixup_json.py $CONFIG_DIR/$THIS_CONFIG.json
-  if [ "$IS_WIN" == 1 ]; then
-    # Use Linux/UNIX line endings
-    dos2unix $CONFIG_DIR/$THIS_CONFIG.json
-  fi
+  echo "fixup file: $THIS_CONFIG"
+  ./$CONFIG_DIR/fixup_json.py $THIS_CONFIG
 done
 
 #  The symlinks are no longer needed after generating the .json files.
 if [ -L third_party/libwebrtc/buildtools ]; then
   rm third_party/libwebrtc/buildtools
@@ -137,17 +74,9 @@
 
 # After all the json files are generated they are used to generate moz.build files.
 echo "Building moz.build files from gn json files"
 ./mach build-backend -b GnMozbuildWriter --verbose
 
-# Make sure all the moz.build files have unix line endings if generated
-# on Windows.
-if [ "$IS_WIN" == 1 ]; then
-  MODIFIED_BUILD_FILES=`hg status --modified --added --no-status --include '**/moz.build'`
-  for BUILD_FILE in $MODIFIED_BUILD_FILES
-  do
-    dos2unix $BUILD_FILE
-  done
-fi
+rm -rf $MOZ_OBJDIR
 
 echo
 echo "Done generating gn build files. You should now be able to build with ./mach build"
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_android.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_android.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_android.mozconfig
+++ /dev/null
@@ -1,18 +0,0 @@
-# Use the same Java binary that was specified in bootstrap. This way, if the default system
-# Java is different than what Firefox needs, users should just need to override it (with
-# $JAVA_HOME) when running bootstrap, rather than when interacting with the build.
-ac_add_options --with-java-bin-path=/usr/lib/jvm/java-8-openjdk-amd64/bin
-# Build GeckoView/Firefox for Android:
-ac_add_options --enable-application=mobile/android
-# Targeting the following architecture.
-# For regular phones, no --target is needed.
-# For x86 emulators (and x86 devices, which are uncommon):
-# ac_add_options --target=i686
-# For newer phones.
-ac_add_options --target=aarch64
-# For x86_64 emulators (and x86_64 devices, which are even less common):
-# ac_add_options --target=x86_64
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_arm64_android
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_False.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_False.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_False.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_False.json
@@ -27718,12 +27718,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_False.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_False.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_False.mozconfig
+++ /dev/null
@@ -1,6 +0,0 @@
-ac_add_options --target=aarch64
-ac_add_options --enable-bootstrap
-ac_add_options --enable-default-toolkit=cairo-gtk3-wayland-only
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_arm64_linux_False
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_True.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_True.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_True.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_True.json
@@ -27972,12 +27972,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_True.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_True.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_linux_True.mozconfig
+++ /dev/null
@@ -1,5 +0,0 @@
-ac_add_options --target=aarch64
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_arm64_linux_True
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_mac.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_mac.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_mac.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_mac.json
@@ -25329,11 +25329,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [],
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_mac.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_mac.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_mac.mozconfig
+++ /dev/null
@@ -1,4 +0,0 @@
-ac_add_options --target=aarch64
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_arm64_mac
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_win.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_win.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_win.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_win.json
@@ -41743,11 +41743,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_win.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_win.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm64_win.mozconfig
+++ /dev/null
@@ -1,3 +0,0 @@
-ac_add_options --target=aarch64
-mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-x64_False_arm64_win
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm_android.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm_android.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_arm_android.mozconfig
+++ /dev/null
@@ -1,18 +0,0 @@
-# Use the same Java binary that was specified in bootstrap. This way, if the default system
-# Java is different than what Firefox needs, users should just need to override it (with
-# $JAVA_HOME) when running bootstrap, rather than when interacting with the build.
-ac_add_options --with-java-bin-path=/usr/lib/jvm/java-8-openjdk-amd64/bin
-# Build GeckoView/Firefox for Android:
-ac_add_options --enable-application=mobile/android
-# Targeting the following architecture.
-# For regular phones, no --target is needed.
-# For x86 emulators (and x86 devices, which are uncommon):
-# ac_add_options --target=i686
-# For newer phones.
-# ac_add_options --target=aarch64
-# For x86_64 emulators (and x86_64 devices, which are even less common):
-# ac_add_options --target=x86_64
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_arm_android
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_android.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_android.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_android.mozconfig
+++ /dev/null
@@ -1,18 +0,0 @@
-# Use the same Java binary that was specified in bootstrap. This way, if the default system
-# Java is different than what Firefox needs, users should just need to override it (with
-# $JAVA_HOME) when running bootstrap, rather than when interacting with the build.
-ac_add_options --with-java-bin-path=/usr/lib/jvm/java-8-openjdk-amd64/bin
-# Build GeckoView/Firefox for Android:
-ac_add_options --enable-application=mobile/android
-# Targeting the following architecture.
-# For regular phones, no --target is needed.
-# For x86 emulators (and x86 devices, which are uncommon):
-# ac_add_options --target=i686
-# For newer phones.
-# ac_add_options --target=aarch64
-# For x86_64 emulators (and x86_64 devices, which are even less common):
-ac_add_options --target=x86_64
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_x64_android
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_False.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_False.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_False.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_False.json
@@ -27715,12 +27715,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_False.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_False.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_False.mozconfig
+++ /dev/null
@@ -1,5 +0,0 @@
-ac_add_options --enable-bootstrap
-ac_add_options --enable-default-toolkit=cairo-gtk3-wayland-only
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_x64_linux_False
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_True.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_True.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_True.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_True.json
@@ -27971,12 +27971,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_True.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_True.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_linux_True.mozconfig
+++ /dev/null
@@ -1,4 +0,0 @@
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_x64_linux_True
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_mac.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_mac.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_mac.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_mac.json
@@ -24832,11 +24832,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [],
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_mac.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_mac.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_mac.mozconfig
+++ /dev/null
@@ -1,2 +0,0 @@
-mk_add_options MOZ_OBJDIR=obj-x64_False_x64_mac
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_openbsd.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_openbsd.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_openbsd.mozconfig
+++ /dev/null
@@ -1,11 +0,0 @@
-mk_add_options PYTHON=/usr/local/bin/python3.9
-export M4=/usr/local/bin/gm4
-export CC=/usr/local/bin/clang
-export CXX=/usr/local/bin/clang++
-ac_add_options --with-libclang-path=/usr/local/lib
-
-ac_add_options --without-wasm-sandboxed-libraries
-
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_x64_openbsd
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_win.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_win.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_win.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_win.json
@@ -41619,11 +41619,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_win.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_win.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_win.mozconfig
+++ /dev/null
@@ -1,2 +0,0 @@
-mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-x64_False_x64_win
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_android.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_android.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_android.mozconfig
+++ /dev/null
@@ -1,18 +0,0 @@
-# Use the same Java binary that was specified in bootstrap. This way, if the default system
-# Java is different than what Firefox needs, users should just need to override it (with
-# $JAVA_HOME) when running bootstrap, rather than when interacting with the build.
-ac_add_options --with-java-bin-path=/usr/lib/jvm/java-8-openjdk-amd64/bin
-# Build GeckoView/Firefox for Android:
-ac_add_options --enable-application=mobile/android
-# Targeting the following architecture.
-# For regular phones, no --target is needed.
-# For x86 emulators (and x86 devices, which are uncommon):
-ac_add_options --target=i686
-# For newer phones.
-# ac_add_options --target=aarch64
-# For x86_64 emulators (and x86_64 devices, which are even less common):
-# ac_add_options --target=x86_64
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_x86_android
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_False.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_False.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_False.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_False.json
@@ -28189,12 +28189,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_False.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_False.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_False.mozconfig
+++ /dev/null
@@ -1,6 +0,0 @@
-ac_add_options --target=i686
-ac_add_options --enable-bootstrap
-ac_add_options --enable-default-toolkit=cairo-gtk3-wayland-only
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_x86_linux_False
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_True.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_True.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_True.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_True.json
@@ -28445,12 +28445,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_True.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_True.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_linux_True.mozconfig
+++ /dev/null
@@ -1,5 +0,0 @@
-ac_add_options --target=i686
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_False_x86_linux_True
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_win.json b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_win.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_win.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_win.json
@@ -41619,11 +41619,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_win.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_win.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_win.mozconfig
+++ /dev/null
@@ -1,3 +0,0 @@
-ac_add_options --target=i686
-mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-x64_False_x86_win
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_android.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_android.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_android.mozconfig
+++ /dev/null
@@ -1,19 +0,0 @@
-# Use the same Java binary that was specified in bootstrap. This way, if the default system
-# Java is different than what Firefox needs, users should just need to override it (with
-# $JAVA_HOME) when running bootstrap, rather than when interacting with the build.
-ac_add_options --with-java-bin-path=/usr/lib/jvm/java-8-openjdk-amd64/bin
-# Build GeckoView/Firefox for Android:
-ac_add_options --enable-application=mobile/android
-# Targeting the following architecture.
-# For regular phones, no --target is needed.
-# For x86 emulators (and x86 devices, which are uncommon):
-# ac_add_options --target=i686
-# For newer phones.
-ac_add_options --target=aarch64
-# For x86_64 emulators (and x86_64 devices, which are even less common):
-# ac_add_options --target=x86_64
-ac_add_options --enable-debug
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_arm64_android
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_False.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_False.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_False.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_False.json
@@ -27483,12 +27483,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_False.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_False.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_False.mozconfig
+++ /dev/null
@@ -1,7 +0,0 @@
-ac_add_options --enable-debug
-ac_add_options --target=aarch64
-ac_add_options --enable-bootstrap
-ac_add_options --enable-default-toolkit=cairo-gtk3-wayland-only
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_arm64_linux_False
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_True.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_True.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_True.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_True.json
@@ -27737,12 +27737,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_True.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_True.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_linux_True.mozconfig
+++ /dev/null
@@ -1,6 +0,0 @@
-ac_add_options --enable-debug
-ac_add_options --target=aarch64
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_arm64_linux_True
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_mac.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_mac.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_mac.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_mac.json
@@ -24859,11 +24859,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [],
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_mac.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_mac.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_mac.mozconfig
+++ /dev/null
@@ -1,5 +0,0 @@
-ac_add_options --enable-debug
-ac_add_options --target=aarch64
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_arm64_mac
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_win.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_win.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_win.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_win.json
@@ -41038,11 +41038,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_win.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_win.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm64_win.mozconfig
+++ /dev/null
@@ -1,4 +0,0 @@
-ac_add_options --enable-debug
-ac_add_options --target=aarch64
-mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-x64_True_arm64_win
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm_android.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm_android.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_arm_android.mozconfig
+++ /dev/null
@@ -1,19 +0,0 @@
-# Use the same Java binary that was specified in bootstrap. This way, if the default system
-# Java is different than what Firefox needs, users should just need to override it (with
-# $JAVA_HOME) when running bootstrap, rather than when interacting with the build.
-ac_add_options --with-java-bin-path=/usr/lib/jvm/java-8-openjdk-amd64/bin
-# Build GeckoView/Firefox for Android:
-ac_add_options --enable-application=mobile/android
-# Targeting the following architecture.
-# For regular phones, no --target is needed.
-# For x86 emulators (and x86 devices, which are uncommon):
-# ac_add_options --target=i686
-# For newer phones.
-# ac_add_options --target=aarch64
-# For x86_64 emulators (and x86_64 devices, which are even less common):
-# ac_add_options --target=x86_64
-ac_add_options --enable-debug
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_arm_android
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_android.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_android.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_android.mozconfig
+++ /dev/null
@@ -1,19 +0,0 @@
-# Use the same Java binary that was specified in bootstrap. This way, if the default system
-# Java is different than what Firefox needs, users should just need to override it (with
-# $JAVA_HOME) when running bootstrap, rather than when interacting with the build.
-ac_add_options --with-java-bin-path=/usr/lib/jvm/java-8-openjdk-amd64/bin
-# Build GeckoView/Firefox for Android:
-ac_add_options --enable-application=mobile/android
-# Targeting the following architecture.
-# For regular phones, no --target is needed.
-# For x86 emulators (and x86 devices, which are uncommon):
-# ac_add_options --target=i686
-# For newer phones.
-# ac_add_options --target=aarch64
-# For x86_64 emulators (and x86_64 devices, which are even less common):
-ac_add_options --target=x86_64
-ac_add_options --enable-debug
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_x64_android
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_False.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_False.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_False.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_False.json
@@ -27478,12 +27478,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_False.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_False.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_False.mozconfig
+++ /dev/null
@@ -1,6 +0,0 @@
-ac_add_options --enable-debug
-ac_add_options --enable-bootstrap
-ac_add_options --enable-default-toolkit=cairo-gtk3-wayland-only
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_x64_linux_False
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_True.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_True.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_True.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_True.json
@@ -27734,12 +27734,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_True.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_True.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_linux_True.mozconfig
+++ /dev/null
@@ -1,5 +0,0 @@
-ac_add_options --enable-debug
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_x64_linux_True
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_mac.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_mac.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_mac.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_mac.json
@@ -24358,11 +24358,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [],
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_mac.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_mac.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_mac.mozconfig
+++ /dev/null
@@ -1,4 +0,0 @@
-ac_add_options --enable-debug
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_x64_mac
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_openbsd.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_openbsd.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_openbsd.mozconfig
+++ /dev/null
@@ -1,12 +0,0 @@
-mk_add_options PYTHON=/usr/local/bin/python3.9
-export M4=/usr/local/bin/gm4
-export CC=/usr/local/bin/clang
-export CXX=/usr/local/bin/clang++
-ac_add_options --with-libclang-path=/usr/local/lib
-
-ac_add_options --without-wasm-sandboxed-libraries
-
-ac_add_options --enable-debug
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_x64_openbsd
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_win.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_win.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_win.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_win.json
@@ -40671,11 +40671,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_win.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_win.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_win.mozconfig
+++ /dev/null
@@ -1,3 +0,0 @@
-ac_add_options --enable-debug
-mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-x64_True_x64_win
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_android.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_android.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_android.mozconfig
+++ /dev/null
@@ -1,19 +0,0 @@
-# Use the same Java binary that was specified in bootstrap. This way, if the default system
-# Java is different than what Firefox needs, users should just need to override it (with
-# $JAVA_HOME) when running bootstrap, rather than when interacting with the build.
-ac_add_options --with-java-bin-path=/usr/lib/jvm/java-8-openjdk-amd64/bin
-# Build GeckoView/Firefox for Android:
-ac_add_options --enable-application=mobile/android
-# Targeting the following architecture.
-# For regular phones, no --target is needed.
-# For x86 emulators (and x86 devices, which are uncommon):
-ac_add_options --target=i686
-# For newer phones.
-# ac_add_options --target=aarch64
-# For x86_64 emulators (and x86_64 devices, which are even less common):
-# ac_add_options --target=x86_64
-ac_add_options --enable-debug
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_x86_android
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_False.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_False.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_False.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_False.json
@@ -27715,12 +27715,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_False.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_False.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_False.mozconfig
+++ /dev/null
@@ -1,7 +0,0 @@
-ac_add_options --enable-debug
-ac_add_options --target=i686
-ac_add_options --enable-bootstrap
-ac_add_options --enable-default-toolkit=cairo-gtk3-wayland-only
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_x86_linux_False
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_True.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_True.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_True.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_True.json
@@ -27971,12 +27971,10 @@
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/third_party/libwebrtc/third_party/pipewire/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
-                "/third_party/pipewire/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_True.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_True.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_linux_True.mozconfig
+++ /dev/null
@@ -1,6 +0,0 @@
-ac_add_options --enable-debug
-ac_add_options --target=i686
-ac_add_options --enable-bootstrap
-
-mk_add_options MOZ_OBJDIR=obj-x64_True_x86_linux_True
-
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_win.json b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_win.json
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_win.json
+++ b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_win.json
@@ -40671,11 +40671,10 @@
                 "//third_party/abseil-cpp/absl/strings:strings"
             ],
             "include_dirs": [
                 "/media/libyuv/libyuv/include/",
                 "/media/libyuv/",
-                "/media/libyuv/libyuv/include/",
                 "/third_party/pipewire/",
                 "//",
                 "//third_party/abseil-cpp/"
             ],
             "libs": [
diff --git a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_win.mozconfig b/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_win.mozconfig
deleted file mode 100644
--- a/dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_win.mozconfig
+++ /dev/null
@@ -1,4 +0,0 @@
-ac_add_options --target=i686
-ac_add_options --enable-debug
-mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-x64_True_x86_win
-
diff --git a/python/mozbuild/mozbuild/gn_processor.py b/python/mozbuild/mozbuild/gn_processor.py
--- a/python/mozbuild/mozbuild/gn_processor.py
+++ b/python/mozbuild/mozbuild/gn_processor.py
@@ -155,19 +155,31 @@
 def filter_gn_config(gn_result, config, sandbox_vars, input_vars, gn_target):
     # Translates the raw output of gn into just what we'll need to generate a
     # mozbuild configuration.
     gn_out = {"targets": {}, "sandbox_vars": sandbox_vars, "gn_gen_args": input_vars}
 
-    gn_mozbuild_vars = (
-        "MOZ_DEBUG",
-        "OS_TARGET",
-        "HOST_CPU_ARCH",
-        "CPU_ARCH",
-        "MOZ_X11",
-    )
+    cpus = {
+        "arm64": "aarch64",
+        "x64": "x86_64",
+    }
+    oses = {
+        "android": "Android",
+        "linux": "Linux",
+        "mac": "Darwin",
+        "openbsd": "OpenBSD",
+        "win": "WINNT",
+    }
 
-    mozbuild_args = {k: config.substs.get(k) for k in gn_mozbuild_vars}
+    mozbuild_args = {
+        "MOZ_DEBUG": "1" if input_vars.get("is_debug") else None,
+        "OS_TARGET": oses[input_vars["target_os"]],
+        "HOST_CPU_ARCH": cpus.get(input_vars["host_cpu"], input_vars["host_cpu"]),
+        "CPU_ARCH": cpus.get(input_vars["target_cpu"], input_vars["target_cpu"]),
+    }
+    if input_vars["target_os"] in ("linux", "android", "openbsd"):
+        mozbuild_args["MOZ_X11"] = "1" if input_vars.get("use_x11") else None
+
     gn_out["mozbuild_args"] = mozbuild_args
     all_deps = find_deps(gn_result["targets"], gn_target)
 
     for target_fullname in all_deps:
         raw_spec = gn_result["targets"][target_fullname]
@@ -578,12 +590,36 @@
     def str_for_arg(v):
         if v in (True, False):
             return str(v).lower()
         return '"%s"' % v
 
+    gn_input_variables = input_variables.copy()
+    gn_input_variables.update(
+        {
+            "concurrent_links": 1,
+            "action_pool_depth": 1,
+        }
+    )
+
+    if input_variables["target_os"] == "win":
+        gn_input_variables.update(
+            {
+                "visual_studio_path": "/",
+                "visual_studio_version": 2015,
+                "wdk_path": "/",
+            }
+        )
+    if input_variables["target_os"] == "mac":
+        gn_input_variables.update(
+            {
+                "mac_sdk_path": "/",
+                "enable_wmax_tokens": False,
+            }
+        )
+
     gn_args = "--args=%s" % " ".join(
-        ["%s=%s" % (k, str_for_arg(v)) for k, v in six.iteritems(input_variables)]
+        ["%s=%s" % (k, str_for_arg(v)) for k, v in six.iteritems(gn_input_variables)]
     )
     # Don't make use_x11 part of the string for openbsd to avoid creating
     # new json files.
     gn_arg_string = "_".join(
         [
@@ -620,20 +656,47 @@
             if not gn_binary:
                 raise Exception(
                     "The GN program must be present to generate GN configs."
                 )
 
-            generate_gn_config(
-                obj.config,
-                mozpath.join(obj.srcdir, obj.target_dir),
-                mozpath.join(obj.objdir, obj.target_dir),
-                obj.non_unified_sources,
-                gn_binary,
-                obj.gn_input_variables,
-                obj.gn_sandbox_variables,
-                obj.gn_target,
-            )
+            vars_set = []
+            for is_debug in (True, False):
+                for target_os in ("android", "linux", "mac", "openbsd", "win"):
+                    target_cpus = ["x64"]
+                    if target_os in ("android", "linux", "mac", "win"):
+                        target_cpus.append("arm64")
+                    if target_os == "android":
+                        target_cpus.append("arm")
+                    if target_os in ("android", "linux", "win"):
+                        target_cpus.append("x86")
+                    for target_cpu in target_cpus:
+                        vars = {
+                            "host_cpu": "x64",
+                            "is_debug": is_debug,
+                            "target_cpu": target_cpu,
+                            "target_os": target_os,
+                        }
+                        if target_os == "linux":
+                            for use_x11 in (True, False):
+                                vars["use_x11"] = use_x11
+                                vars_set.append(vars.copy())
+                        else:
+                            if target_os == "openbsd":
+                                vars["use_x11"] = True
+                            vars_set.append(vars)
+
+            for vars in vars_set:
+                generate_gn_config(
+                    obj.config,
+                    mozpath.join(obj.srcdir, obj.target_dir),
+                    mozpath.join(obj.objdir, obj.target_dir),
+                    obj.non_unified_sources,
+                    gn_binary,
+                    vars,
+                    obj.gn_sandbox_variables,
+                    obj.gn_target,
+                )
         return True
 
     def consume_finished(self):
         pass
 
diff --git a/third_party/libwebrtc/build/config/BUILDCONFIG.gn b/third_party/libwebrtc/build/config/BUILDCONFIG.gn
--- a/third_party/libwebrtc/build/config/BUILDCONFIG.gn
+++ b/third_party/libwebrtc/build/config/BUILDCONFIG.gn
@@ -234,11 +234,11 @@
 } else if (target_os == "fuchsia") {
   _default_toolchain = "//build/toolchain/fuchsia:$target_cpu"
 } else if (target_os == "ios") {
   _default_toolchain = "//build/toolchain/mac:ios_clang_$target_cpu"
 } else if (target_os == "mac") {
-  assert(host_os == "mac", "Mac cross-compiles are unsupported.")
+  #assert(host_os == "mac", "Mac cross-compiles are unsupported.")
   _default_toolchain = "//build/toolchain/mac:clang_$target_cpu"
 } else if (target_os == "win") {
   # On Windows, we use the same toolchain for host and target by default.
   # Beware, win cross builds have some caveats, see docs/win_cross.md
   if (is_clang) {
diff --git a/third_party/libwebrtc/build/config/mac/mac_sdk.gni b/third_party/libwebrtc/build/config/mac/mac_sdk.gni
--- a/third_party/libwebrtc/build/config/mac/mac_sdk.gni
+++ b/third_party/libwebrtc/build/config/mac/mac_sdk.gni
@@ -84,15 +84,15 @@
     rebase_path("$root_build_dir/sdk/xcode_links", "//"),
   ]
 }
 sdk_info_args += [ mac_sdk_name ]
 
-_mac_sdk_result = exec_script(script_name, sdk_info_args, "scope")
-xcode_version = _mac_sdk_result.xcode_version
-xcode_build = _mac_sdk_result.xcode_build
-machine_os_build = _mac_sdk_result.machine_os_build
-machine_os_build_major = _mac_sdk_result.machine_os_build_major
+#_mac_sdk_result = exec_script(script_name, sdk_info_args, "scope")
+#xcode_version = _mac_sdk_result.xcode_version
+#xcode_build = _mac_sdk_result.xcode_build
+#machine_os_build = _mac_sdk_result.machine_os_build
+#machine_os_build_major = _mac_sdk_result.machine_os_build_major
 if (mac_sdk_path == "" && use_system_xcode && ios_use_goma_rbe) {
   mac_sdk_path = _mac_sdk_result.sdk_path
 }
 
 if (use_system_xcode) {
diff --git a/third_party/libwebrtc/build/config/win/visual_studio_version.gni b/third_party/libwebrtc/build/config/win/visual_studio_version.gni
--- a/third_party/libwebrtc/build/config/win/visual_studio_version.gni
+++ b/third_party/libwebrtc/build/config/win/visual_studio_version.gni
@@ -33,7 +33,7 @@
 } else {
   assert(visual_studio_version != "",
          "You must set the visual_studio_version if you set the path")
   assert(wdk_path != "",
          "You must set the wdk_path if you set the visual studio path")
-  visual_studio_runtime_dirs = []
+  visual_studio_runtime_dirs = ""
 }
diff --git a/third_party/libwebrtc/build/toolchain/mac/BUILD.gn b/third_party/libwebrtc/build/toolchain/mac/BUILD.gn
--- a/third_party/libwebrtc/build/toolchain/mac/BUILD.gn
+++ b/third_party/libwebrtc/build/toolchain/mac/BUILD.gn
@@ -13,11 +13,11 @@
   import("//build/config/ios/ios_sdk.gni")
 }
 import("//build/config/mac/mac_sdk.gni")
 import("//build/config/mac/symbols.gni")
 
-assert(host_os == "mac")
+#assert(host_os == "mac")
 
 import("//build/toolchain/cc_wrapper.gni")
 import("//build/toolchain/concurrent_links.gni")
 import("//build/toolchain/goma.gni")
 import("//build/toolchain/toolchain.gni")
diff --git a/third_party/libwebrtc/build/win/BUILD.gn b/third_party/libwebrtc/build/win/BUILD.gn
--- a/third_party/libwebrtc/build/win/BUILD.gn
+++ b/third_party/libwebrtc/build/win/BUILD.gn
@@ -17,12 +17,12 @@
   ]
 }
 
 build_with_mozilla = false
 if (is_win && !build_with_mozilla) {
-  assert(host_os != "mac" || target_cpu != "x86",
-         "Windows cross-builds from Mac must be 64-bit.")
+#  assert(host_os != "mac" || target_cpu != "x86",
+#         "Windows cross-builds from Mac must be 64-bit.")
 
   action("copy_cdb_to_output") {
     script = "//build/win/copy_cdb_to_output.py"
     inputs = [
       script,
diff --git a/third_party/libwebrtc/build_overrides/build.gni b/third_party/libwebrtc/build_overrides/build.gni
--- a/third_party/libwebrtc/build_overrides/build.gni
+++ b/third_party/libwebrtc/build_overrides/build.gni
@@ -45,10 +45,14 @@
          "Do not allow building targets with the default " +
              "hermetic toolchain if the minimum OS version is not met.")
   use_system_xcode = _result == 0
 }
 
+use_system_xcode = false
+xcode_version = "10.15"
+mac_xcode_version = "default"
+
 declare_args() {
   # WebRTC doesn't depend on //base from production code but only for testing
   # purposes. In any case, it doesn't depend on //third_party/perfetto which
   # is used for base tracing, so this feature is disabled.
   enable_base_tracing = false
diff --git a/third_party/libwebrtc/modules/desktop_capture/desktop_capture_generic_gn/moz.build b/third_party/libwebrtc/modules/desktop_capture/desktop_capture_generic_gn/moz.build
--- a/third_party/libwebrtc/modules/desktop_capture/desktop_capture_generic_gn/moz.build
+++ b/third_party/libwebrtc/modules/desktop_capture/desktop_capture_generic_gn/moz.build
@@ -23,12 +23,14 @@
 
 LOCAL_INCLUDES += [
     "!/ipc/ipdl/_ipdlheaders",
     "/ipc/chromium/src",
     "/media/libyuv/",
+    "/media/libyuv/libyuv/include/",
     "/third_party/libwebrtc/",
     "/third_party/libwebrtc/third_party/abseil-cpp/",
+    "/third_party/pipewire/",
     "/tools/profiler/public"
 ]
 
 UNIFIED_SOURCES += [
     "/third_party/libwebrtc/modules/desktop_capture/blank_detector_desktop_capturer_wrapper.cc",
@@ -71,16 +73,10 @@
     DEFINES["_LIBCPP_HAS_NO_ALIGNED_ALLOCATION"] = True
     DEFINES["__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES"] = "0"
     DEFINES["__STDC_CONSTANT_MACROS"] = True
     DEFINES["__STDC_FORMAT_MACROS"] = True
 
-    LOCAL_INCLUDES += [
-        "/media/libyuv/libyuv/include/",
-        "/media/libyuv/libyuv/include/",
-        "/third_party/pipewire/"
-    ]
-
     UNIFIED_SOURCES += [
         "/third_party/libwebrtc/modules/desktop_capture/mac/desktop_configuration_monitor.cc",
         "/third_party/libwebrtc/modules/desktop_capture/mac/full_screen_mac_application_handler.cc",
         "/third_party/libwebrtc/modules/desktop_capture/mac/window_list_utils.cc"
     ]
@@ -100,17 +96,10 @@
     DEFINES["_LARGEFILE64_SOURCE"] = True
     DEFINES["_LARGEFILE_SOURCE"] = True
     DEFINES["__STDC_CONSTANT_MACROS"] = True
     DEFINES["__STDC_FORMAT_MACROS"] = True
 
-    LOCAL_INCLUDES += [
-        "/media/libyuv/libyuv/include/",
-        "/media/libyuv/libyuv/include/",
-        "/third_party/pipewire/",
-        "/third_party/pipewire/"
-    ]
-
     OS_LIBS += [
         "dl",
         "rt"
     ]
 
@@ -138,15 +127,10 @@
     DEFINES["_LARGEFILE64_SOURCE"] = True
     DEFINES["_LARGEFILE_SOURCE"] = True
     DEFINES["__STDC_CONSTANT_MACROS"] = True
     DEFINES["__STDC_FORMAT_MACROS"] = True
 
-    LOCAL_INCLUDES += [
-        "/media/libyuv/libyuv/include/",
-        "/third_party/pipewire/"
-    ]
-
     OS_LIBS += [
         "X11",
         "Xcomposite",
         "Xdamage",
         "Xext",
@@ -202,16 +186,10 @@
     DEFINES["_UNICODE"] = True
     DEFINES["_WIN32_WINNT"] = "0x0A00"
     DEFINES["_WINDOWS"] = True
     DEFINES["__STD_C"] = True
 
-    LOCAL_INCLUDES += [
-        "/media/libyuv/libyuv/include/",
-        "/media/libyuv/libyuv/include/",
-        "/third_party/pipewire/"
-    ]
-
     OS_LIBS += [
         "crypt32",
         "d3d11",
         "dxgi",
         "iphlpapi",

